public/app/plugins/datasource/prometheus/querybuilder/components/metrics-modal/state/helpers.ts

```diff
@@ -32,7 +33,16 @@ export async function setMetrics(
   let metricsData: MetricsData | undefined;
 
   metricsData = initialMetrics?.map((m: string) => {
-    const type = getMetadataType(m, datasource.languageProvider.metricsMetadata!);
+    let type = getMetadataType(m, datasource.languageProvider.metricsMetadata!);
+    let inferredType = false;
+    if (!type && inferType) {
+      type = metricTypeHints(m);
+
+      if (type) {
+        inferredType = true;
+      }
+    }
```

Comment:
In this code block, wouldn't the `inferredType` always be false? I think it should have been defined outside of the `map` callback. Or am I wrong?

Reply:
Same code is in `helpers.ts` too.

Reply:
```
    let type = getMetadataType(m, datasource.languageProvider.metricsMetadata!);
    // inferredType will be set on each metric data element, and it will always be false unless the setting (inferType) is turned on and a type has been inferred
    let inferredType = false;
    // inferType is the setting and if turned on, then we check each metric and update the inferredType in the metric data
    if (!type && inferType) {
      type = metricTypeHints(m);

      if (type) {
        inferredType = true;
      }
    }
```

Reply:
I can add some comments to this code to clarify what is happening. 

`inferType` is the state setting to use the inferring logic in `metricTypeHints`
If it is turned off, each metric data element will be given the inferredType field of false.

If `inferType` setting is true, then we perform the logic for each metric. 
If we set `inferredType` outside the map, if a metric does not have a type, if then inferred a type, the flag is set true, and the next metric does have a type, the flag will still be set true and that metric with an actual type will be given a field that it was inferred. 

So, we need to set that default to false for each metric inside the map.

Reply:
Yeah, you are right I missed some part of the code.